custom-k8s-metrics-demo/README.md

```diff
@@ -0,0 +1,120 @@
+# Pixie Custom Kubernetes Metrics Demo
+
+Autoscale the number of pods in your Kubernetes deployment based on request throughput, without any code changes.
+
+## What is this demo?
+
+This demo provides an example implementation of a custom Kubernetes metric from [Pixie](https://github.com/pixie-io/pixie) data. Specifically, it provides a metric for the number of HTTP requests per second by Kubernetes pod. This custom metric can then be used as an input to a [Horizontal Pod Autoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) in order to to scale up/down the number of pods.
+
+This demo was based off of the example in [kubernetes-sigs/custom-metrics-apiserver].
+
+## Prerequisites
+
+* If you don't already have one, set up a [Kubernetes cluster](https://docs.px.dev/installing-pixie/setting-up-k8s/)
+* [Install Pixie](https://docs.px.dev/installing-pixie/install-guides/) to your Kubernetes cluster
+* [Optional] Install [hey](https://github.com/rakyll/hey) for testing the demo application 
+
+## Usage
+
+1. Create a secret containing the Pixie API credentials for your Kubernetes cluster:
+
+```
+# Get your current cluster name from your Kubernetes context
+kubectl config current-context
+# Get the Pixie Cluster ID for the above cluster name.
+# Record the value of the `ID` column for this cluster
+px get viziers
+
+# Create an API key
+# Record the value of the `Key` parameter
+px api-key create
+
+kubectl -n px-custom-metrics create secret generic px-credentials --from-literal=px-api-key=<YOUR API KEY VALUE HERE> --from-literal=px-cluster-id=<YOUR CLUSTER ID VALUE HERE>
+```
+
+2. [Optional] If using self-hosted Pixie Cloud, update PX_CLOUD_ADDR in `px-custom-metrics.yaml`.
```

Comment:
It looks like even if you're using non-hosted Pixie cloud, you need to set this:
```
2021/10/07 20:09:35 `PX_CLOUD_ADDR` is not set.
```